library/build.gradle

```diff
@@ -26,33 +26,41 @@ jar {
     }
 }
 
+wrapper {
+    gradleVersion = '7.3.1'
+}
+
 dependencies {
+    def fuelVersion = '2.3.1'
+    def log4jVersion = '2.15.0'
+
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8"
     implementation "org.jetbrains.kotlinx:kotlinx-coroutines-core:1.3.2"
 
-    //core
-    implementation 'com.github.kittinunf.fuel:fuel:2.2.1'
+    // https://mvnrepository.com/artifact/com.github.kittinunf.fuel/fuel
+    implementation "com.github.kittinunf.fuel:fuel:${fuelVersion}"
+    implementation "com.github.kittinunf.fuel:fuel-coroutines:${fuelVersion}"
 
-    implementation 'org.slf4j:slf4j-log4j12:1.7.5'
-    implementation 'io.github.microutils:kotlin-logging:1.7.7'
+    // https://mvnrepository.com/artifact/org.apache.logging.log4j/log4j-slf4j-impl
+    implementation "org.apache.logging.log4j:log4j-slf4j-impl:${log4jVersion}"
```

Comment:
Actually! this still pulls in `log4j-slf4j-impl` and `log4j` into the final artifact (fat jar)... which is contrary to the practices suggested by slf4j (as the idea of slf4j is that actual implementation should be plugged in at deployment time) 🤔

Reply:
Would it make sense to mark log4j-slf4j-impl as e.g. optional or compile-time-only dependency?

Reply:
Since if anybody uses sth different than log4j (e.g. logback)... then they'll still have log4j on their classpath, with no easy option for exclusion (since the log4j classes are baked into the fat jar)